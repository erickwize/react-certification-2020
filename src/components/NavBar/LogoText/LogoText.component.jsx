import React from 'react';
import imagen from './tube.png';
import './LogoText.styles.css';
import { useHistory } from 'react-router';

export default function LogoText(){
    const history = useHistory();

    const goToHome = () => {
        history.push('/');
    }

    return(
        <span onClick={goToHome} className="logoBox">
            <h3>Wize</h3>
            <img src={imagen} alt="imagen"/>
        </span>
    );
}