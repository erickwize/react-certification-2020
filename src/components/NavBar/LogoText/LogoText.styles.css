.logoBox{
    width: 20%;
    height: 7vh;
    display: flex;
    align-items: center;
    margin: auto;
}

.logoBox h3{
    color: #f0efef;
    line-height: 7vh;
    font-size: 33px;
    font-family: 'Roboto', sans-serif;
}

.logoBox img{
    width: 50px;
    margin-left: 3px;
}

@media only screen and (max-width: 650px) {
    .logoBox{
        display: none;
    }
}