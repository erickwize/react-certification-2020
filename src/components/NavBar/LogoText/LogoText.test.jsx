import React from 'react';
import {render, screen} from '@testing-library/react';
import LogoText from './LogoText.component';

describe('Logo Text', () => {
    it('renders', () => {
        render( <LogoText/> );
        const wizePart = screen.getByRole('heading');
        const TubePart = screen.getByAltText('imagen');
        expect(wizePart).toBeInTheDocument();
        expect(TubePart).toBeInTheDocument();   
    });

    it('renders correctly', () => {
        render( <LogoText/> );
        const wizePart = screen.getByRole('heading');
        const TubePart = screen.getByAltText('imagen');
        expect(wizePart.textContent).toBe("Wize");
        expect(TubePart.src.endsWith("tube.png")).toBeTruthy();
    });

    it('renders correctly', () => {
        render( <LogoText/> );
        const wizePart = screen.getByRole('heading');
        const TubePart = screen.getByAltText('imagen');
        expect(wizePart).toMatchSnapshot();
        expect(TubePart).toMatchSnapshot();
    });
})