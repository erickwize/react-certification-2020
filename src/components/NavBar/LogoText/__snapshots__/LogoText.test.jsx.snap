// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Text renders correctly 1`] = `
<h3>
  Wize
</h3>
`;

exports[`Logo Text renders correctly 2`] = `
<img
  alt="imagen"
  src="tube.png"
/>
`;
